src/main/java/me/steinborn/krypton/mixin/shared/network/flushconsolidation/ThreadedAnvilChunkStorageMixin.java

```diff
@@ -64,60 +101,124 @@ public void updatePosition(ServerPlayerEntity player) {
             this.updateWatchedSection(player);
-
+            
             if (!isWatchingWorld) {
                 this.ticketManager.handleChunkLeave(oldPos, player);
             }
-
+            
             if (!noChunkGen) {
                 this.ticketManager.handleChunkEnter(newPos, player);
             }
-
+            
             if (!isWatchingWorld && noChunkGen) {
                 this.playerChunkWatchingManager.disableWatch(player);
             }
-
+            
             if (isWatchingWorld && !noChunkGen) {
                 this.playerChunkWatchingManager.enableWatch(player);
             }
-
+            
             long oldChunkPos = ChunkPos.toLong(oldPos.getX(), oldPos.getZ());
             long newChunkPos = ChunkPos.toLong(newPos.getX(), newPos.getZ());
             this.playerChunkWatchingManager.movePlayer(oldChunkPos, newChunkPos, player);
-
-            // If the player is in the same world as this tracker, we should send them chunks.
-            if (this.world == player.world) {
-                this.sendChunks(oldPos, player);
+        }
+        
+        // The player *always* needs to be send chunks, as for some reason both chunk loading & unloading packets are handled
+        // by the same method (why mojang)
+        if (player.world == this.world) {
+            this.sendChunkWatchPackets(oldPos, player);
+        }
+    }
+    
+    @Inject(method = "tickEntityMovement", at = @At("HEAD"))
+    public void disableAutoFlushForEntityTracking(CallbackInfo info) {
+        for (ServerPlayerEntity player : world.getPlayers()) {
+            AutoFlushUtil.setAutoFlush(player, false);
+        }
+    }
+    
+    @Inject(method = "tickEntityMovement", at = @At("RETURN"))
+    public void enableAutoFlushForEntityTracking(CallbackInfo info) {
+        for (ServerPlayerEntity player : world.getPlayers()) {
+            AutoFlushUtil.setAutoFlush(player, true);
+        }
+    }
+    
+    /**
+     * @param player                The player
+     * @param pos                   The position of the chunk to send
+     * @param mutableObject         A new mutable object
+     * @param oldWithinViewDistance If the chunk was previously within the player's view distance
+     * @param newWithinViewDistance If the chunk is now within the player's view distance
+     */
+    @Shadow
+    public abstract void sendWatchPackets(ServerPlayerEntity player, ChunkPos pos, MutableObject<ChunkDataS2CPacket> mutableObject,
+                                          boolean oldWithinViewDistance, boolean newWithinViewDistance);
+    
+    @Shadow
+    protected abstract boolean doesNotGenerateChunks(ServerPlayerEntity player);
+    
+    @Shadow
+    protected abstract ChunkSectionPos updateWatchedSection(ServerPlayerEntity serverPlayerEntity);
+    
+    /**
+     * Sends watch packets to the client in a spiral for a player which has *no* chunks loaded in the area.
+     *
+     * @author solonovamax
```

Comment:
not really your code here (and not even mine), I'd just remove this

Reply:
I added the javadoc for the method description, intellij added the `@author` automatically. I'll remove it.